sklearn/cluster/_k_means_elkan.pyx

```diff
@@ -314,19 +321,23 @@ def elkan_iter_chunked_dense(
                 weight_in_clusters_chunk,
                 update_centers)
 
-        # reduction from local buffers. The gil is necessary for that to avoid
+        # reduction from local buffers. The lock is necessary for that to avoid
```

Comment:
```suggestion
        # reduction from local buffers.
```
Please put the second part of the comment further below and rephrase it a little. Why lock instead of nogil, etc.

Reply:
done. Let me know what you think.

Reply:
I like it 👍